File: api/src/main/java/fr/huiitre/tools/modules/core/notification/application/usecase/GetMyNotificationsUseCase.java
package fr.huiitre.tools.modules.core.notification.application.usecase;

import java.util.List;

import org.springframework.stereotype.Service;

import fr.huiitre.tools.modules.core.notification.api.view.NotificationView;
import fr.huiitre.tools.modules.core.notification.application.port.NotificationRepository;
import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.core.security.application.ports.CurrentUserProvider;
import fr.huiitre.tools.modules.core.security.application.usecase.SecuredUseCase;

@Service
public class GetMyNotificationsUseCase implements SecuredUseCase {

    private final NotificationRepository notificationRepository;
    private final CurrentUserProvider currentUserProvider;

    public GetMyNotificationsUseCase(
            NotificationRepository notificationRepository,
            CurrentUserProvider currentUserProvider) {
        this.notificationRepository = notificationRepository;
        this.currentUserProvider = currentUserProvider;
    }

    @Override
    public RoleCode requiredRole() {
        return RoleCode.READ_ONLY;
    }

    public List<NotificationView> execute() {
        Long userId = Long.parseLong(currentUserProvider.getCurrentUserId());

        return notificationRepository.findActiveForUser(userId).stream()
                .map(NotificationView::fromDomain)
                .toList();
    }
}

File: api/Modules/Notifications/Application/Usecases/GetMyNotificationsUseCase.cs
using Tools.ApiCore.Modules.Notifications.Application.Ports;
using Tools.ApiCore.Modules.Notifications.Application.Views;
using Tools.ApiCore.Modules.Security.Application.Services;
using Tools.ApiCore.Modules.Security.Application.Usecases;

namespace Tools.ApiCore.Modules.Notifications.Application.Usecases;

// Notifications de l'appelant.
//
// Ni `RequiredRole` ni `RequiredModule` ne sont surchargés : le contrôle par défaut suffit, soit
// un compte authentifié portant au moins READ_ONLY. Lire ses propres notifications n'exige rien
// de plus — c'est le rôle qu'exigeait déjà l'API Java.
public sealed class GetMyNotificationsUseCase(
    UseCaseAuthorizer authorizer,
    INotificationRepository notificationRepository) : SecuredUseCase(authorizer)
{
    // Le destinataire n'est pas un argument : c'est l'appelant validé, hors de portée du client.
    public Task<IReadOnlyList<NotificationView>> Execute()
    {
        return notificationRepository.FindActiveForUserAsync(CurrentUser.UserId);
    }
}
